CMakeLists.txt

```diff
@@ -75,6 +81,142 @@ target_compile_definitions(notcurses
     FORTIFY_SOURCE=2 _GNU_SOURCE SOURCE_DEFAULT
 )
 
+# libnotcurses++
+set(NCPP_SOURCES
+  src/libcpp/NotCurses.cc
+  src/libcpp/PanelReel.cc
+  src/libcpp/Plane.cc
+  src/libcpp/Root.cc
+  src/libcpp/Tablet.cc
+  src/libcpp/Visual.cc
+  )
+
+add_library(
+  notcurses++
+  SHARED
+  ${NCPP_SOURCES}
+  )
+
+add_library(
+  notcurses++-static
+  STATIC
+  ${NCPP_SOURCES}
+  )
+set_target_properties(
+  notcurses++-static PROPERTIES
+  OUTPUT_NAME notcurses++
+  )
+
+set_target_properties(
+  notcurses++ PROPERTIES
+  VERSION ${PROJECT_VERSION}
+  SOVERSION ${PROJECT_VERSION_MAJOR}
+  OUTPUT_NAME "notcurses++")
+
+set(NCPP_INCLUDE_DIRS
+  include
+    "${PROJECT_BINARY_DIR}/include"
+    "${TERMINFO_INCLUDE_DIRS}"
+  )
+
+target_include_directories(notcurses++
+  PRIVATE ${NCPP_INCLUDE_DIRS}
+  )
+
+target_include_directories(notcurses++-static
+  PRIVATE ${NCPP_INCLUDE_DIRS}
+  )
+
+target_link_libraries(notcurses++
+  PRIVATE
+  notcurses)
+
+set(NCPP_COMPILE_OPTIONS
+  -Wall
+  -Wextra
+  -W
+  -Wshadow
+  -Wformat
+  -Werror=format-security
+  -Wnull-dereference
+  -Wmisleading-indentation
+  -Wunused
+  -Wpedantic
+  -Wsuggest-override
+  -Wno-c99-extensions
+  -fno-strict-aliasing
+  -ffunction-sections
+  -funswitch-loops
+  -finline-limit=300
+  -fstack-protector
+  -fno-rtti
+  -fno-exceptions
+  -mpopcnt
```

Comment:
All of this seems a bit...extra :). And `-mpopcnt` surely doesn't belong here. Aren't all the `-m` options specifying things explicitly about the local machine (in this case, the existence of a population count instruction?) x86 didn't have POPCOUNT until SSE4.2, and ARM still doesn't have it AFAIK.

Reply:
Oh, `-mpopcnt` I something I missed (I have a set of standard flags I use for local builds across projects), yeah, it doesn't belong in the generic build script. However, the `-f*` stuff is actually either important or useful IMO.

Reply:
OK, sounds good then.